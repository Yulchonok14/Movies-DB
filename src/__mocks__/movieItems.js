export const movieItems = [
    {
        id: 111,
        title: 'Bloodshot',
        release_date: '2020-05-01',
        vote_average: 3.6,
        genres: ['Action', 'Drama', 'Sci-Fi'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 110
    },
    {
        id: 222,
        title: 'The invisible man',
        release_date: '2019-05-01',
        vote_average: 4.6,
        genres: ['Horror', 'Sci-Fi', 'Thriller'],
        poster_path: 'https://avatars.mds.yandex.net/get-kinopoisk-image/1900788/a90dd0dd-a8ae-4112-9865-23eb8f5879a8/360',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 185
    },
    {
        id: 333,
        title: 'Monster trucks',
        release_date: '2017-06-01',
        vote_average: 3.9,
        genres: ['Adventure', 'Action', 'Comedy'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/5/5e/Monster_Trucks_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 154
    },
    {
        id: 444,
        title: 'Cold comes the night',
        release_date: '2018-05-01',
        vote_average: 4.4,
        genres: ['Drama', 'Thriller', 'Crime'],
        poster_path: 'https://laughingsquid.com/wp-content/uploads/2014/01/CCTN_poster_Billy_HD-for-web.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 129
    },
    {
        id: 555,
        title: 'Snowden',
        release_date: '2020-05-21',
        vote_average: 4.8,
        genres: ['Drama', 'Thriller', 'Biography'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/1/1a/Snowden_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 113
    },
    {
        id: 666,
        title: 'Knock knock',
        release_date: '2011-05-01',
        vote_average: 2.6,
        genres: ['Horror', 'Thriller', 'Adventure'],
        poster_path: 'https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 187
    }
];