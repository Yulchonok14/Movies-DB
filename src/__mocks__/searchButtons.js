export const searchByButtonsArr = [
    {id: 111, text: 'title', label: 'title', isActive: true},
    {id: 222, text: 'genres', label: 'genre', isActive: false}
];