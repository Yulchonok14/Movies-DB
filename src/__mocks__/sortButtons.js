export const sortByButtonsArr = [
    {id: 111, text: 'release_date', label: 'release date', isActive: true},
    {id: 222, text: 'vote_average', label: 'rating', isActive: false}
];