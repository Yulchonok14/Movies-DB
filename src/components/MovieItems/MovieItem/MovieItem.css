.MovieItem {
    width: auto;
    max-width: 380px;
    margin-bottom: 50px;
    padding: 0 30px;
    text-align: center;

    box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
    transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.MovieItem:hover {
    border-top-left-radius: 10px;
    border-bottom-left-radius: 10px;
    box-shadow: 0 14px 28px rgba(255,255,255,255.85), 0 10px 10px rgba(255,255,255,255.82);
}


.MovieItem img {
    width: 100%;
}

.Description {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    color: #FFFFFF;
    margin: 25px 0;
    text-align: left;
}

.Description div {
    display: flex;
    flex-direction: column;
    width: 65%
 }

.Description div span {
    margin-bottom: 10px;
}

.Date {
    border: 1px solid rgba(255, 255, 255, 0.3);
    padding: 5px 15px;
    border-radius: 4px;
    margin-left: 5px;
}

.Genre {
    font-size: small;
    opacity: 0.5;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    width: 100%;
}

@media (min-width: 600px) and (max-width: 1000px) {
    .MovieItem {
        width: 35vw;
        max-width: 380px;
    }
}

@media (min-width: 1001px) {
    .MovieItem {
        width: 25vw;
        max-width: 380px;
    }
}