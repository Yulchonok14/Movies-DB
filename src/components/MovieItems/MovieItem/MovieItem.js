import React from 'react';
import PropTypes from 'prop-types';

import classes from './MovieItem.css';
import ImageNotFound from '../../../assets/images/image_not_found.png';

const MovieItem = (props) => (
    <div onClick={props.clicked} className={classes.MovieItem}>
        <img src={props.imgUrl} onError={(e)=>{e.target.onerror = null; e.target.src=ImageNotFound}}/>
        <div className={classes.Description}>
            <div>
                <span>{props.title}</span>
                <span className={classes.Genre}>{props.genre}</span>
            </div>
            <span className={classes.Date}>{props.date}</span>
        </div>
    </div>
);

MovieItem.propTypes = {
    clicked: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired
};

export default MovieItem;