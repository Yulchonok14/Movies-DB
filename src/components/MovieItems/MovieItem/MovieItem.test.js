import React from 'react';

import { configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import MovieItem from './MovieItem';
import { movieItems } from '../../../__mocks__/movieItems';
import ImageNotFound from '../../../assets/images/image_not_found.png';

configure({
    adapter: new Adapter()
});
describe('<MovieItem/>', () => {
    let wrapper;
    const mockCallBack = jest.fn();
    const props = {
        key: movieItems[5].id,
        imgUrl: movieItems[5].poster_path,
        title: movieItems[5].title,
        date: movieItems[5].release_date,
        genre: movieItems[5].genres.join(', '),
        clicked: mockCallBack
    };

    beforeEach(() => {
        wrapper = mount(<MovieItem {...props}/>);
    });

    it('MovieItem should have image and onError handler', () => {
        expect(wrapper.find('img').props())
            .toEqual({
                src: movieItems[5].poster_path,
                onError: expect.any(Function)
            });
    });

    it('MovieItem renders correctly', () => {
        const tree = renderer
            .create(<MovieItem {...props}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call any passed in onError after an image load error', () => {
        const mEvent = {target: wrapper.getDOMNode()};
        wrapper.find('img').prop('onError')(mEvent);
        expect(mEvent.target.src).toBe(ImageNotFound);
    });
})
;