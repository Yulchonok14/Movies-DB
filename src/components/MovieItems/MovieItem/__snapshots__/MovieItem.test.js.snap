// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem/> MovieItem renders correctly 1`] = `
<div
  onClick={[MockFunction]}
>
  <img
    onError={[Function]}
    src="https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg"
  />
  <div>
    <div>
      <span>
        Knock knock
      </span>
      <span>
        Horror, Thriller, Adventure
      </span>
    </div>
    <span>
      2011-05-01
    </span>
  </div>
</div>
`;
